### PR TITLE
[Merged by Bors] - feat(RingTheory/PowerSeries/WellKnown): changed `PowerSeries.invOneSubPow`

### PR DESCRIPTION
**Pull request by FMLJohn**
The definition of `PowerSeries.invOneSubPow` has been changed into the following:
Given a commutative ring `S` and a number `d : ℕ`, `PowerSeries.invOneSubPow S d` is the multiplicative inverse of `(1 - X) ^ d` in `S⟦X⟧ˣ`. When `d` is `0`, `PowerSeries.invOneSubPow S d` will just be `1`. When `d` is positive, `PowerSeries.invOneSubPow S d` will be `∑ n, Nat.choose (d - 1 + n) (d - 1)`.

### CODE REVIEW

File: Mathlib/RingTheory/PowerSeries/WellKnown.lean [hunk 1]
```diff
@@ -97,39 +98,69 @@ theorem mk_one_pow_eq_mk_choose_add :
...
 -/
 theorem invOneSubPow_eq_inv_one_sub_pow :
-    invOneSubPow d = (Units.mkOfMulEqOne (1 - X) (mk 1 : S⟦X⟧)
-    <| Eq.trans (mul_comm _ _) mk_one_mul_one_sub_eq_one)⁻¹ ^ (d + 1) := by
-  rw [inv_pow]
-  exact (DivisionMonoid.inv_eq_of_mul _ (invOneSubPow d) <| by
-    rw [← Units.val_eq_one, Units.val_mul, Units.val_pow_eq_pow_val]
-    exact (invOneSubPow d).inv_val).symm
+    invOneSubPow S d = (Units.mkOfMulEqOne (1 - X) (mk 1 : S⟦X⟧)
+    <| Eq.trans (mul_comm _ _) (mk_one_mul_one_sub_eq_one S))⁻¹ ^ d := by
```
Comment (Ruben-VandeVelde) [inline Mathlib/RingTheory/PowerSeries/WellKnown.lean:141]: More readable as
```suggestion
    invOneSubPow S d =
      (Units.mkOfMulEqOne (1 - X) (mk 1 : S⟦X⟧) <|
        Eq.trans (mul_comm _ _) (mk_one_mul_one_sub_eq_one S))⁻¹ ^ d := by
```

File: Mathlib/RingTheory/PowerSeries/WellKnown.lean [hunk 2]
```diff
@@ -97,39 +98,69 @@ theorem mk_one_pow_eq_mk_choose_add :
...
+  induction d with
+  | zero => exact Eq.symm <| pow_zero _
+  | succ d _ =>
+      rw [inv_pow]
+      refine (DivisionMonoid.inv_eq_of_mul _ (invOneSubPow S (d + 1)) <| by
+        rw [← Units.val_eq_one, Units.val_mul, Units.val_pow_eq_pow_val]
+        exact (invOneSubPow S (d + 1)).inv_val).symm
```
Comment (Ruben-VandeVelde) [inline Mathlib/RingTheory/PowerSeries/WellKnown.lean:148]: Either
```suggestion
      refine (DivisionMonoid.inv_eq_of_mul _ (invOneSubPow S (d + 1)) ?_).symm
      rw [← Units.val_eq_one, Units.val_mul, Units.val_pow_eq_pow_val]
      exact (invOneSubPow S (d + 1)).inv_val
```
or
```suggestion
      exact (DivisionMonoid.inv_eq_of_mul _ (invOneSubPow S (d + 1)) <| by
        rw [← Units.val_eq_one, Units.val_mul, Units.val_pow_eq_pow_val]
        exact (invOneSubPow S (d + 1)).inv_val).symm
```

**Review by Ruben-VandeVelde — COMMENTED** [review]
Some suggestions

File: Mathlib/RingTheory/PowerSeries/WellKnown.lean [hunk 3]
```diff
@@ -97,39 +98,68 @@ theorem mk_one_pow_eq_mk_choose_add :
...
+  | d + 1 => {
+    val := mk fun n => Nat.choose (d + n) d
+    inv := (1 - X) ^ (d + 1)
+    val_inv := by
+      rw [← mk_one_pow_eq_mk_choose_add, ← mul_pow, mk_one_mul_one_sub_eq_one, one_pow]
+    inv_val := by
+      rw [← mk_one_pow_eq_mk_choose_add, ← mul_pow, mul_comm, mk_one_mul_one_sub_eq_one, one_pow]
+    }
+
+theorem invOneSubPow_zero_val_eq_one : (invOneSubPow S 0).val = 1 := by
```
Comment (YaelDillies) [inline Mathlib/RingTheory/PowerSeries/WellKnown.lean:117]:
```suggestion
theorem invOneSubPow_zero : invOneSubPow S 0 = 1 := by
```

**Review by YaelDillies — APPROVED** [review]
maintainer merge

apart from that. Thanks!

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Comment by jcommelin** [comment]
bors r-

bors d+

**Comment by jcommelin** [comment]
Sorry, I clicked the wrong button :see_no_evil:

**Comment by FMLJohn** [comment]
bors merge

**Pull request closed without merging**